src/lib/RegistrationLogic.sol

```diff
@@ -54,7 +56,9 @@ contract RegistrationLogic is ConstructorLogic {
         internal
         returns (bytes32 claimHash)
     {
-        return _deriveClaimHashAndRegisterCompact(sponsor, typehash, 0x120, _domainSeparator(), sponsorSignature);
```

Comment:
we should do this branchlessly inside of `_deriveClaimHashAndRegisterCompact` IMHO